File: backend/java-springboot-hive/src/main/java/com/sakrafux/realworld/domain/model/Comment.java
package com.sakrafux.realworld.domain.model;

import com.sakrafux.realworld.domain.model.Profile;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;

/**
 * Domain model representing a Comment on an Article.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Comment {
    private Long id;
    private ZonedDateTime createdAt;
    private ZonedDateTime updatedAt;
    private String body;
    private Profile author;
}

File: backend/typescript-node-express-hive/src/cells/comment/comment.ts
export interface Author {
    username: string;
    bio: string;
    image: string | null;
    following: boolean;
}

export class Comment {
    constructor(
        public id: number,
        public createdAt: Date,
        public updatedAt: Date,
        public body: string,
        public author: Author,
    ) {}
}
